### PR TITLE
Rework reflect Flags

### PR DESCRIPTION
**Pull request by nicolasstucki**
* Add `Deferred`, `Infix`, `JavaStatic`, `Method`, `NoInit` and `Transparent`
* Remove `ModuleClass`
* Rename `Object` to `Module`

### CODE REVIEW

File: library/src/scala/quoted/Quotes.scala [hunk 1]
```diff
@@ -3903,6 +3905,9 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       /** Is this symbol a type parameter marked as covariant `+` */
       def Covariant: Flags
 
+      /** Is a declared, but not defined member */
```
Comment (liufengyun) [inline library/src/scala/quoted/Quotes.scala:3908]:
```suggestion
      /** Is this symbol a declared, but not defined member */
```

File: library/src/scala/quoted/Quotes.scala [hunk 2]
```diff
@@ -3933,12 +3938,18 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       /** Is this symbol `implicit` */
       def Implicit: Flags
 
+      /** Is an infix method or type */
```
Comment (liufengyun) [inline library/src/scala/quoted/Quotes.scala:3941]:
```suggestion
      /** Is this symbol an infix method or type */
```

File: library/src/scala/quoted/Quotes.scala [hunk 3]
```diff
@@ -3933,12 +3938,18 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
...
+      def Infix: Flags
+
       /** Is this symbol `inline` */
       def Inline: Flags
 
       /** Is this symbol defined in a Java class */
       def JavaDefined: Flags
 
+      /** Is implemented as a Java static */
```
Comment (liufengyun) [inline library/src/scala/quoted/Quotes.scala:3950]:
```suggestion
      /** Is this symbol implemented as a Java static */
```

File: library/src/scala/quoted/Quotes.scala [hunk 4]
```diff
@@ -4001,6 +4014,10 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
 
       /** Is this symbol a trait */
       def Trait: Flags
+
+      /** Is a transparent inline method or trait */
```
Comment (liufengyun) [inline library/src/scala/quoted/Quotes.scala:4018]:
```suggestion
      /** Is this symbol a transparent inline method or trait */
```

**Review by liufengyun — APPROVED** [review]
LGTM

**Comment by nicolasstucki** [comment]
I will align the documentation with `dotc.core.Flags` in a separate PR.

File: library/src/scala/quoted/Quotes.scala [hunk 5]
```diff
@@ -3888,7 +3888,9 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       /** Is this symbol `abstract` */
       def Abstract: Flags
 
-      /** Was this symbol generated by Scala compiler */
+      /** Is this generated by Scala compiler.
```
Comment (smarter) [inline library/src/scala/quoted/Quotes.scala:3891]:
```suggestion
      /** Is this symbol compiler-generated and not meant to be called in source code.
```

**Pull request merged**